File: 91. Decode Ways/91.go
package leetcode

import(
	"strconv"
)

func numDecodings(s string) int {
    if len(s) == 0{
		return 0
	}

	v := 0
	w := 1
	p := "0"
	mys := strings.Split(s, "")

	for _,letter := range mys{
		tmp := 0
		if letter != "0"{
			tmp += w
		}
		ge, _ := strconv.Atoi(letter)
		shi, _ := strconv.Atoi(p)
		value := ge + shi*10
		if value>9 && value<27{
			tmp += v
		}
		v = w
		w = tmp
		p = letter
	}

	return w
}

File: 91. Decode Ways/91.c


int numDecodings(char * s){
    if(*s == 0){
        return 0;
    }

    int w = 1;
    int v = 0;
    char p = '0';

    while (*s != 0){
        int tmp = 0;
        if(*s != '0'){
            tmp += w;
        }

        int shi = p-'0';
        int ge = *s-'0';
        int value = shi*10+ge;

        if(value>9 && value<27){
            tmp += v;
        }

        v = w;
        w = tmp;
        p = *s;
        s += 1;
    }

    return w;
}

